internal/controller/release_controller.go

```diff
@@ -200,7 +201,7 @@ func (r *ReleaseReconciler) ensureManagement(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	mgmtObj.Spec.Providers = hmc.GetDefaultProviders()
+	mgmtObj.Spec.Providers = providers.List()
```

Comment:
File `internal/providers/azure.go` is still needed due to a need of populating this list on first start of `manager`.

Reply:
One of the approaches that can allow removal of mentioned file is taken in https://github.com/k0rdent/kcm/pull/833/files

Specifically: 
- [YAML provider loader](https://github.com/k0rdent/kcm/blob/38de0406237caea10af8c8dc7e5a2fd4db1dbeaa/internal/providers/yaml.go), just without Lua parts
- [Actual YAML provider definition](https://github.com/k0rdent/kcm/blob/38de0406237caea10af8c8dc7e5a2fd4db1dbeaa/providers/azure.yml), also ignore Lua parts here
- [YAML loader](https://github.com/k0rdent/kcm/blob/38de0406237caea10af8c8dc7e5a2fd4db1dbeaa/cmd/main.go#L82-L85) in `manager` start

Reply:
Or any other way that can populate list with extra values during `manager` start

Or maybe we could rethink logic behind that list altogether